homeassistant/components/iron_os/number.py

```diff
@@ -30,15 +43,39 @@
 class IronOSNumberEntityDescription(NumberEntityDescription):
     """Describes IronOS number entity."""
 
-    value_fn: Callable[[LiveDataResponse], float | int | None]
-    max_value_fn: Callable[[LiveDataResponse], float | int]
+    value_fn: Callable[[LiveDataResponse, SettingsDataResponse], float | int | None]
+    max_value_fn: Callable[[LiveDataResponse], float | int] | None = None
     set_key: CharSetting
+    set_fn: Callable[[float], float | int] | None = None
```

Comment:
We should rather rename this to something like `transform_new_value_fn` or something. Although that one isn't ideal either, `set_fn` kinda leaves the idea that the function can be used for setting the value while it does not

Reply:
there is `value_fn`, that converts to a humand friendly value, the lambda for converting back could be `raw_value_fn`? Is that better?